harmony/aws.py

```diff
@@ -54,17 +55,21 @@ def _get_aws_client(config, service):
         The configuration for the current runtime environment.
     service : string
         The AWS service name for which to construct a client, e.g. "s3" or "sqs"
+    user_agent : string
+        The user agent that is requesting the aws service.
+        E.g. harmony/0.0.0 harmony-sit
 
     Returns
     -------
     s3_client : boto3.*.Client
         A client appropriate for accessing the provided service
     """
+    boto_cfg = Config(user_agent=user_agent)
```

Comment:
In looking at the boto3 documentation I see they also have a `user_agent_extra` field. https://botocore.amazonaws.com/v1/documentation/api/latest/reference/config.html

I think what we'd want to do is append the harmony agent to whatever the existing user agent string is, so we should use `user_agent_extra` instead. I'm curious what others think we should do.

Reply:
Good point. Using user_agent_extra would give us more information in the logs.

Reply:
That begs the question of whether I should do the same for the 'requests' library (in http.py) if possible.

Reply:
First, nice work, especially for a first ticket.  I have some overall review comments, that I think end up related to this thread (i.e. where to get which information from).

I'm not sure we want to make the user agent as optional and easily override-able as it is.  My hunch is that existing services and perhaps new services will never set it.  Those that do set it could set it inconsistently with one another, negating much of the purpose.

I think it could work as follows:
* Keep the existing code almost entirely (slightly modify docs according to new behavior).
* Pass the user agent as env var from Harmony to the Docker image.
* If there is no user agent parameter passed to the lib, default to using the env var.  If there is no env var set, default to "harmony-unknown" as the main Harmony service does.
* If there _is_ a user agent parameter passed to the lib, treat it as `user_agent_extra`, i.e. append it to the Harmony user agent.
* Wholly unnecessary nice-to-have: inject harmony-service-lib/VERSION into the user agent

Reply:
So something like `{boto_or_requests_lib_agent} {harmony_agent} {harmony_service_lib_agent} {agent_passed_to_lib}` where `agent_passed_to_lib` is the optional user agent passed to the harmony service lib and `harmony_agent` can be either the env var passed from harmony or "harmony/version harmony-unknown" when no env var is passed? E.g. `boto-agent/03.2 harmony/0.0.0 harmony-unknown harmony-service-lib/4.0 gdal/1.0`.

Regarding the env var -- originally we were thinking of repurposing the harmony env var `harmonyClientId` to pass the user agent from harmony to the service lib (https://github.com/nasa/harmony/pull/8/files), as it is always gets passed along via the message.client field. So instead, I should leave this alone, and introduce a new env var? Just making sure I understand.

Reply:
Updates have been implemented per our slack convo. The user-agent now looks more like what was pasted in the above comment, with the harmony_agent coming in as an env variable from Harmony frontend (rather than relying on the Message). The boto call was also updated to pass the full user agent string to user_agent_extra rather than overriding its default.